_pytest/config.py

```diff
@@ -442,12 +442,14 @@ def import_plugin(self, modname):
         try:
             __import__(importspec)
         except ImportError as e:
-            new_exc = ImportError('Error importing plugin "%s": %s' % (modname, safe_str(e.args[0])))
-            # copy over name and path attributes
-            for attr in ('name', 'path'):
-                if hasattr(e, attr):
-                    setattr(new_exc, attr, getattr(e, attr))
-            raise new_exc
+            new_exc_type = ImportError
+            new_exc_message = 'Error importing plugin "%s": %s' % (modname, safe_str(e.args[0]))
+
+            # Raising the original traceback with a more informative exception
```

Comment:
please take a look at using six.reraise/six.raise_from

Reply:
Agreed, `six.reraise` seems to be more appropriate here ([`six.raises_from`](https://pythonhosted.org/six/#six.raise_from) is the same as `raise new_exc` in Python 2).

We want to preserve the original traceback as much as possible, but also want to include which plugin was the cause of the exception. After playing around this code seems to do the trick:

```python
import sys
import six

def do_import(name):
    raise ValueError('some bizarre error not related to the plugin name at all')

def import_some_plugin(name):
    do_import(name)

def import_plugins():

    try:
        import_some_plugin("myplugin")
    except Exception as e:
        new_e = type(e)(str(e) + '\n' + 'Error importing pytest plugin: "myplugin"')
        six.reraise(type(e), new_e, sys.exc_info()[2])

import_plugins()
```

I get in Python 2.7:

```
.tmp\foo.py:18: in <module>
    import_plugins()
.tmp\foo.py:16: in import_plugins
    six.reraise(type(e), new_e, sys.exc_info()[2])
.tmp\foo.py:13: in import_plugins
    import_some_plugin("myplugin")
.tmp\foo.py:8: in import_some_plugin
    do_import(name)
.tmp\foo.py:5: in do_import
    raise ValueError('some bizarre error not related to the plugin name at all')
E   ValueError: some bizarre error not related to the plugin name at all
E   Error importing pytest plugin: "myplugin"
!!!!!!!!!!!!!!!!!!! Interrupted: 1 errors during collection !!!!!!!!!!!!!!!!!!!
```

And Python 3.6:

```
Traceback (most recent call last):
  File "C:/pytest/.tmp/foo.py", line 17, in import_plugins
    import_some_plugin("myplugin")
  File "C:/pytest/.tmp/foo.py", line 11, in import_some_plugin
    do_import(name)
  File "C:/pytest/.tmp/foo.py", line 7, in do_import
    raise ValueError('some bizarre error not related to the plugin name at all')
ValueError: some bizarre error not related to the plugin name at all

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:/pytest/.tmp/foo.py", line 23, in <module>
    import_plugins()
  File "C:/pytest/.tmp/foo.py", line 20, in import_plugins
    six.reraise(type(e), new_e, sys.exc_info()[2])
  File "C:\pytest\.env36\lib\site-packages\six.py", line 685, in reraise
    raise value.with_traceback(tb)
  File "C:/pytest/.tmp/foo.py", line 17, in import_plugins
    import_some_plugin("myplugin")
  File "C:/pytest/.tmp/foo.py", line 11, in import_some_plugin
    do_import(name)
  File "C:/pytest/.tmp/foo.py", line 7, in do_import
    raise ValueError('some bizarre error not related to the plugin name at all')
ValueError: some bizarre error not related to the plugin name at all
Error importing pytest plugin: "myplugin"
```

What do you think @RonnyPfannschmidt? In Python 3 we might consider not adding the extra message.